backend/src/daemon/daemon_thread.rs

```diff
@@ -113,8 +119,12 @@ impl ThreadClient {
         });
         // XXX let caller know it was canceled?
         match receiver.await {
-            Ok(Ok(res)) => res,
-            _ => Ok(()),
+            Ok(Ok(res)) => res.and_then(|x| {
+                serde_json::from_str(&x).map_err(|err| {
+                    format!("failed to deserialize: {}\n{:#?}", err, err)
+                })
+            }),
+            err => Err(format!("receiver.await returned {:?}", err)),
```

Comment:
This will make it print an error if the operation has been cancelled.

A lot of the current complexity in `daemon_thread.rs` is to handle cancellation of operations when (for instance) multiple sets to the same color index are issued, so it can't queue up an arbitrary number of operations that reverse each other.